From 81149344a5ba53b30e8ab7d77d605dc484a0a3ff Mon Sep 17 00:00:00 2001
From: Dylan Thacker-Smith <Dylan.Smith@shopify.com>
Date: Tue, 31 Mar 2020 10:19:25 -0400
Subject: [PATCH] Fix ParseTreeVisitorTest for ruby-head

Fix parse_tree_visitor_test.rb for Ruby 3. Upstream has this for liquid 5.0.0,
patch can be removed when updated to 5.0.0.

---
 test/integration/parse_tree_visitor_test.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Patch-Source: https://src.fedoraproject.org/rpms/rubygem-liquid/raw/f34/f/03-fix-parse-tree-visitor-test-for-ruby-3.patch

diff --git a/test/integration/parse_tree_visitor_test.rb b/test/integration/parse_tree_visitor_test.rb
index d1af123..32dbcf6 100644
--- a/test/integration/parse_tree_visitor_test.rb
+++ b/test/integration/parse_tree_visitor_test.rb
@@ -238,7 +238,7 @@ class ParseTreeVisitorTest < Minitest::Test
   def traversal(template)
     ParseTreeVisitor
       .for(Template.parse(template).root)
-      .add_callback_for(VariableLookup, &:name)
+      .add_callback_for(VariableLookup) { |node| node.name } # rubocop:disable Style/SymbolProc
   end
 
   def visit(template)
-- 
2.29.2

